service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -424,187 +404,68 @@ public static List<String> validateRequestInputsAndOutputs(
-                              e.getMessage()));
-                    } catch (JsonProcessingException e) {
-                      // Should be super rare that jackson cannot convert an object to Json...
-                      String errorMsg =
-                          String.format(
-                              "Failed to convert inputs object to JSON for batch in RunSet %s.",
-                              runSet.runSetId());
-                      log.warn(errorMsg, e);
-                      runStateResponseList.add(
-                          storeRun(
-                              requestedIdToRunId.get(entry.getKey()),
-                              null,
-                              runSet,
-                              entry.getValue().getId(),
-                              SYSTEM_ERROR,
-                              errorMsg + e.getMessage()));
-                    }
-                    return null;
-                  })
-              .filter(inputs -> !Objects.isNull(inputs))
-              .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-
-      if (requestedIdToWorkflowInput.isEmpty()) {
-        return runStateResponseList;
-      }
+    return newRunSet;
+  }
 
-      try {
-        // Submit the workflows and store the Runs to database
-        List<WorkflowIdAndStatus> submitWorkflowBatchResponse =
-            cromwellService.submitWorkflowBatch(
-                rawMethodUrl, requestedIdToWorkflowInput, workflowOptionsJson, userToken);
-
-        runStateResponseList.addAll(
-            submitWorkflowBatchResponse.stream()
-                .map(
-                    idAndStatus -> {
-                      UUID requestedId = UUID.fromString(idAndStatus.getId());
-                      return storeRun(
-                          requestedIdToRunId.get(requestedId),
-                          idAndStatus.getId(),
-                          runSet,
-                          requestedIdToRecord.get(requestedId).getId(),
-                          INITIALIZING,
-                          null);
-                    })
-                .toList());
-      } catch (cromwell.client.ApiException e) {
-        String errorMsg =
-            String.format(
-                "Cromwell submission failed for batch in RunSet %s. ApiException: ",
-                runSet.runSetId());
-        log.warn(errorMsg, e);
-        runStateResponseList.addAll(
-            requestedIdToWorkflowInput.keySet().stream()
-                .map(
-                    requestedId ->
-                        storeRun(
-                            requestedIdToRunId.get(requestedId),
-                            null,
-                            runSet,
-                            requestedIdToRecord.get(requestedId).getId(),
-                            SYSTEM_ERROR,
-                            errorMsg + e.getMessage()))
-                .toList());
+  private List<RunStateResponse> registerRunsInRunSet(
```

Comment:
TOL this might be a good candidate of business logic to move to the RunSetService / RunSetHandler layer